src/main/java/com/clickhouse/kafka/connect/sink/ProxySinkTask.java

```diff
@@ -85,12 +85,23 @@ public void put(final Collection<SinkRecord> records) throws IOException, Execut
         LOGGER.trace(String.format("Got %d records from put API.", records.size()));
         ExecutionTimer processingTime = ExecutionTimer.start();
 
-        Map<String, List<Record>> dataRecords = records.stream()
-                .map(v -> Record.convert(v,
-                        clickHouseSinkConfig.isEnableDbTopicSplit(),
-                        clickHouseSinkConfig.getDbTopicSplitChar(),
-                        clickHouseSinkConfig.getDatabase() ))
-                .collect(Collectors.groupingBy(Record::getTopicAndPartition));
+        Map<String, List<Record>> dataRecords;
+        if (!clickHouseSinkConfig.isExactlyOnce() && clickHouseSinkConfig.isIgnorePartitionsWhenBatching()) {
+            dataRecords = records.stream()
+                    .map(v -> Record.convert(v,
+                            clickHouseSinkConfig.isEnableDbTopicSplit(),
+                            clickHouseSinkConfig.getDbTopicSplitChar(),
+                            clickHouseSinkConfig.getDatabase() ))
+                    .collect(Collectors.groupingBy(Record::getTopic));
```

Comment:
instead of duplicating the code 90-95 and 97 - 102 can we have another method that get clickHouseSinkConfig.isExactlyOnce() && clickHouseSinkConfig.isIgnorePartitionsWhenBatching() as values 
groupByTopicOrTopicAndPartition()